spring-amqp/src/main/java/org/springframework/amqp/support/postprocessor/MessagePostProcessorUtils.java

```diff
@@ -58,6 +61,25 @@ else if (processor instanceof Ordered) {
 		return sorted;
 	}
 
+	/**
+	 * @param collection non nullable collection.
+	 * @param index      the index of the element to be removed.
+	 * @return element that has been removed.
+	 */
+	@Nullable
+	public static MessagePostProcessor remove(Collection<MessagePostProcessor> collection, int index) {
```

Comment:
I am thinking of the remove with an index with the existence of sorting... 

Because also of the following issue:

Say I add 3 Post Processors, and depending on the order, they being shovelled when saved into the instance var `beforePublishPostProcessors`. 

Then I wanted to delete the one at index 2, but because of the sorting shovel, the element at the index 2 is not in place. so it ends up removing another element than what I indented to delete.

Reply:
Wow! Exactly! 

Since we are sorting we definitely can't remove by index afterwards.
Please, drop it off altogether.

Good catch!

Thanks for your sagacity 😄

Reply:
methods removed.